src/apply.c

```diff
@@ -291,7 +291,15 @@ static int apply_binary(
 	git_patch *patch)
 {
 	git_buf reverse = GIT_BUF_INIT;
-	int error;
+	int error = 0;
+
+	if (!patch->binary.old_file.datalen && !patch->binary.new_file.datalen)
+		goto done;
+
+	if (!patch->binary.contains_data) {
+		error = apply_err("patch does not contain binary data");
+		goto done;
+	}
```

Comment:
Yep, you're right.  I did flip these checks, which covered up the fact that I wasn't successfully setting `contains_data = 1` when we were creating a diff from unmodified binary files.  Fixed this up.